lib/entities/bulk-action.ts

```diff
@@ -0,0 +1,72 @@
+import { freezeSys, toPlainObject } from 'contentful-sdk-core'
+import copy from 'fast-copy'
+import {
+  DefaultElements,
+  ISO8601Timestamp,
+  Link,
+  MakeRequest,
+  VersionedLink,
+} from '../common-types'
+
+/** Entities supported by the BulkAction API */
+type Entity = 'Entry' | 'Asset'
+
+export type BulkActionType = 'publish' | 'unpublish' | 'validate'
+
+/** Represents the state of the BulkAction */
+export enum BulkActionStatus {
+  /** BulkAction is pending execution */
+  created = 'created',
+  /** BulkAction has been started and pending completion */
+  inProgress = 'inProgress',
+  /** BulkAction was completed successfully (terminal state) */
+  succeeded = 'succeeded',
+  /** BulkAction failed to complete (terminal state) */
+  failed = 'failed',
+}
+
+interface BulkActionFailedError {
+  sys: { type: 'Error'; id: 'BulkActionFailed' }
+  details: any
+}
+
+export type BulkActionSysProps = {
+  id: string
+  type: 'BulkAction'
+  status: BulkActionStatus
```

Comment:
I would recommend this accepting the raw string values so that people aren't required to use the enum.

```
enum BulkActionStatus {
  ...
}

const STATUSES = Object.values(BulkActionStatus)
type BulkActionStatuses = typeof STATUSES[number]
...

status: BulkActionStatuses
```